### PR TITLE
test(image_projection_based_fusion): add test for image_projection_based_fusion's IoU calculation functions  

### PR DESCRIPTION
**Pull request by SamratThapa120**
## Description

add test for image_projection_based_fusion's IoU calculation functions  
## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/image_projection_based_fusion/test/test_calc_iou_functions.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  roi1.y_offset = 0;
+  roi1.width = 4;
+  roi1.height = 4;
+
+  roi2.x_offset = 2;
+  roi2.y_offset = 0;
+  roi2.width = 4;
+  roi2.height = 4;
+
+  double ioux = calcIoUX(roi1, roi2);
```
Comment (yukkysaito) [inline perception/image_projection_based_fusion/test/test_calc_iou_functions.cpp:71]: ioux doesn't seem to be a common term. what about iou_x?
image

```suggestion
  double iou_x = calcIoUX(roi1, roi2);
```

Comment (kminoda) [reply perception/image_projection_based_fusion/test/test_calc_iou_functions.cpp:71]: @yukkysaito Thanks :pray: 
@SamratThapa120 FYI, try to fix all the `spell-check-partial` error as well (which was originally a required CI, but recently it's not somehow ... :thinking: )

Comment (SamratThapa120) [reply perception/image_projection_based_fusion/test/test_calc_iou_functions.cpp]: fixed

**Review by yukkysaito — APPROVED** [review]
LGTM

**Comment by knzo25** [comment]
@SamratThapa120 
It seems you have a small conflict here. Once you solve it, the PR author is usually the one who merges it. That can be done manually or enabling auto-merge (if this is your first PR, remember that we do squash merges).

**Comment by knzo25** [comment]
(sorry, pressed `close with comment` instead of `comment`)

**Pull request merged**